### DISCUSSION TITLE
Where should I start programming?

### DISCUSSION THREAD

**Discussion by alvagg** [community/community#62057] [Other Feature Feedback, Questions, & Ideas] [answered]
### Hi everyone,

Hello, my name is Álvaro and I am a Telecommunications Engineering student. This year I will start my second year of my degree and I want to learn programming.
The only thing I know about programming is what I've seen so far at the university, which has been 2 programming fundamentals subjects, the first was C and the second focused on object-oriented programming, specifically Java. I already know something about programming thanks to this, but I think that these subjects have helped me more than knowing how to program in these two languages ​​to understand programming fundamentals and concepts since I feel that they have not really focused much on how to make programs and practical things . In the rest of my university degree I will see more programming subjects and I will also have the option of specializing in telematics, so I will learn things at the university for sure.

The point is that I also want to learn on my own and specialize in a specific subject and be able to do practical things and projects. I would like to know what you recommend me to learn, it would be nice if it was something useful and related to telecommunications engineering, and by the way it would be nice to give me advice on how to focus and what to use on the way to learning what you recommend.

I look forward to your recommendations. Thank you so much!

**Comment by CodeMaster7000** [comment, marked as answer]
Hi @alvagg, hope you are having a good day!

Alongside C and Java, you should also aim to learn Python as it can prove to be very useful in focusing on automation. You could also consider Bash since many telecom devices are now built up on Linux environment - so learning Bash gives plenty of advantages in some works. Courses on GeeksforGeeks, Udemy, and RealPython (for Python) can help you along your journey from complete beginner to master.

Good luck with your degree and happy coding!

**Reply by alvagg** [reply]
Thank you very much for the reply! I will consider the recommendations. 
In what order would you recommend looking at each language?
That is to say, do you think it is to look at a little of each one or that it would be better to go one by one and delve into each one individually?

**Reply by CodeMaster7000** [reply]
Initially, I would look into Python a little (then Bash), but once you become familiar with them then it would be useful to delve into each one specifically.
